src/sentry/integrations/slack/message_builder/image_block_builder.py

```diff
@@ -45,13 +44,13 @@ def _build_endpoint_regression_image_block(self) -> SlackBlock | None:
             },
         )
 
-        organization = self.group.organization
-        if self.event is None or self.event.transaction is None or self.event.occurrence is None:
-            return None
-        transaction_name = escape_transaction(self.event.transaction)
-        period = get_relative_time(anchor=get_approx_start_time(self.group), relative_days=14)
-
         try:
+            organization = self.group.organization
+            event = self.group.get_latest_event_for_environments()
```

Comment:
in the future are we going to check for the type of group before doing this?

Reply:
since this is what got us rate limited

Reply:
we are checking the type of group before this function in the `build_image_block` line 32.

Reply:
oh i totally missed that, thanks!